server/src/frameworks/Hardhat/HardhatProject.ts

```diff
@@ -0,0 +1,298 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { ChildProcess, fork } from "child_process";
+import fs from "fs";
+import _ from "lodash";
+import path from "path";
+import { DidChangeWatchedFilesParams } from "vscode-languageserver-protocol";
+import { OpenDocuments, ServerState } from "../../types";
+import { toUnixStyle } from "../../utils";
+import directoryContains from "../../utils/directoryContains";
+import { Logger } from "../../utils/Logger";
+import CompilationDetails from "../base/CompilationDetails";
+import Project from "../base/Project";
+import { LogLevel } from "./worker/WorkerLogger";
+import {
+  BuildCompilationRequest,
+  BuildCompilationResponse,
+  ErrorResponseMessage,
+  FileBelongsRequest,
+  FileBelongsResponse,
+  InitializationFailureMessage,
+  InvalidateBuildCacheMessage,
+  LogMessage,
+  Message,
+  MessageType,
+} from "./worker/WorkerProtocol";
+
+enum WorkerStatus {
+  UNINITIALIZED,
+  INITIALIZING,
+  RUNNING,
+  ERRORED,
+}
+
+const REQUEST_TIMEOUT = 5000;
+
+export default class HardhatProject extends Project {
+  public priority = 2;
+
+  public workerProcess?: ChildProcess;
+
+  public workerStatus = WorkerStatus.UNINITIALIZED;
+
+  public workerLoadFailureReason = "";
+
+  private _onInitialized!: () => void;
+
+  private requestId = 0;
+
+  private _onResponse: { [requestId: number]: (result: any) => void } = {};
+  private _onError: { [requestId: number]: (result: any) => void } = {};
+
+  private logger: Logger;
+
+  private name: string;
+
+  constructor(
+    serverState: ServerState,
+    basePath: string,
+    public configPath: string
+  ) {
+    super(serverState, basePath);
+    this.logger = _.cloneDeep(serverState.logger);
+    this.name = path.basename(basePath);
+    this.logger.tag = `${this.name}`;
+  }
+
+  public id(): string {
+    return this.configPath;
+  }
+
+  public frameworkName(): string {
+    return "Hardhat";
+  }
+
+  public async initialize(): Promise<void> {
+    return new Promise((resolve, _reject) => {
+      this._onInitialized = resolve;
+
+      // Fork WorkerProcess as child process
+      this.workerProcess = fork(
+        path.resolve(__dirname, "worker/WorkerProcess.js"),
+        {
+          cwd: this.basePath,
+          detached: true,
+          execArgv: [],
+        }
+      );
+      this.workerStatus = WorkerStatus.INITIALIZING;
+
+      this.workerProcess.on("message", async (message: Message) => {
+        await this._handleMessage(message);
```

Comment:
Good point. If there are any unhandled errors, it'll log them on the console automatically. But I'll add a try/catch to avoid the `async function without a catch block` part.

Reply:
Consider sentry as well.

A deluge of useless noise, we want to avoid; but if something is going wrong in the gap that might be something we should pay attention to.